src/main/scala/com/databricks/labs/mosaic/core/index/H3IndexSystem.scala

```diff
@@ -32,7 +35,16 @@ object H3IndexSystem extends IndexSystem with Serializable {
       *   Int value representing the resolution.
       */
     override def getResolution(res: Any): Int = {
-        val resolution: Int = res.asInstanceOf[Int]
+        val resolution = (
+          Try(res.asInstanceOf[Int]),
+          Try(res.asInstanceOf[String]),
+          Try(res.asInstanceOf[UTF8String].toString)
+        ) match {
+            case (Success(i), _, _) => i
+            case (_, Success(s), _) => s.toInt
```

Comment:
The toInt should go inside the Try block, otherwise "abc" would not be caught as exception